core/src/main/java/org/mskcc/cbio/portal/scripts/ImportResourceData.java

```diff
@@ -0,0 +1,618 @@
+        while ((line = buff.readLine()) != null) {
+            if (skipLine(line.trim())) {
+                continue;
+            }
+
+            String[] fieldValues = getFieldValues(line, headerNames);
+            addDatum(fieldValues, resources, resourceMap, headerNames);
+        }
+    }
+
+    private boolean skipLine(String line) {
+        return (line.isEmpty() || line.substring(0, 1).equals(METADATA_PREFIX));
+    }
+
+    /**
+     * Takes in the given line and returns the list of field values by splitting the
+     * line on DELIMITER.
+     * 
+     * @param line
+     * @param headerNames
+     * @return the list of values, one for each column. Value will be "" for empty
+     *         columns.
+     */
+    private String[] getFieldValues(String line, String[] headerNames) {
+        // split on delimiter:
+        String[] fieldValues = line.split(DELIMITER, -1);
+
+        // validate: if number of fields is incorrect, give exception
+        if (fieldValues.length != headerNames.length) {
+            throw new IllegalArgumentException("Number of columns in line is not as expected. Expected: "
+                    + headerNames.length + " columns, found: " + fieldValues.length + ", for line: " + line);
+        }
+
+        // now iterate over lines and trim each value:
+        for (int i = 0; i < fieldValues.length; i++) {
+            fieldValues[i] = fieldValues[i].trim();
+        }
+        return fieldValues;
+    }
+
+    private boolean addDatum(String[] fields, List<ResourceDefinition> resources, MultiKeyMap resourceMap, String[] headerNames)
+            throws Exception {
+        int sampleIdIndex = findSampleIdColumn(headerNames);
+        String stableSampleId = (sampleIdIndex >= 0) ? fields[sampleIdIndex] : "";
+        stableSampleId = StableIdUtil.getSampleId(stableSampleId);
+        int patientIdIndex = findPatientIdColumn(headerNames);
+        String stablePatientId = (patientIdIndex >= 0) ? fields[patientIdIndex] : "";
+        stablePatientId = StableIdUtil.getPatientId(stablePatientId);
+        int resourceIdIndex = findResourceIdColumn(headerNames);
+        int urlIndex = findURLColumn(headerNames);
+        int internalSampleId = -1;
+        int internalPatientId = -1;
+
+        // create resource_id set
+        Set<String> patientResourceIdSet = resources.stream().filter(resource -> resource.getResourceType().equals(ResourceDefinition.PATIENT_RESOURCE_TYPE)).map(resource -> resource.getResourceId()).collect(Collectors.toSet());
+        Set<String> sampleResourceIdSet = resources.stream().filter(resource -> resource.getResourceType().equals(ResourceDefinition.SAMPLE_RESOURCE_TYPE)).map(resource -> resource.getResourceId()).collect(Collectors.toSet());
+        Set<String> studyResourceIdSet = resources.stream().filter(resource -> resource.getResourceType().equals(ResourceDefinition.STUDY_RESOURCE_TYPE)).map(resource -> resource.getResourceId()).collect(Collectors.toSet());
```

Comment:
I would break these lines into multiple lines

Reply:
Yeah, that would be better.